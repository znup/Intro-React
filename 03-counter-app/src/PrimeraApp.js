import React from 'react';
import PropTypes from 'prop-types';

// import React, { Fragment } from 'react';

//funcion components

const PrimeraApp = ({ saludo, subtitulo }) => {

  return (
    <>
      <h1>{saludo}</h1>
      {/* no regresa objetos, booleanos */}
      {/* <pre>{JSON.stringify(saludo, null, 3)}</pre> */}
      <p>{ subtitulo }</p>
    </>
  );
  
}

PrimeraApp.propTypes = {
  saludo: PropTypes.string.isRequired
}

PrimeraApp.defaultProps = {
  subtitulo: 'Es un subtitulo'
}

export default PrimeraApp;