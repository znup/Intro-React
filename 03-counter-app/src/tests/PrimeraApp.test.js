import React from "react";
import "@testing-library/jest-dom";
import { shallow } from "enzyme";
import PrimeraApp from "../PrimeraApp";
// import { render } from "@testing-library/react";

describe("Prueba de mi <PrimeraApp />", () => {
  // test("Debe mostrar el mensaje 'Un saludo'", () => {
  //   const saludo = "Un saludo";
  //   const { getByText } = render(<PrimeraApp saludo={saludo} />);
  //   //wrapper producto renderizado que quiero mostrar
  //   expect(getByText(saludo)).toBeInTheDocument();
  // });

  test("Debe mostrar <PrimeraApp /> correctamente", () => {
    const saludo = "Un saludo mas";
    const wrapper = shallow(<PrimeraApp saludo={saludo} />);

    expect(wrapper).toMatchSnapshot();
  });

  test("Debe de mostrar el subtitulo enviado por props", () => {
    const saludo = "Un saludo mas  ";
    const subTitulo = "Es un subtitulo";

    const wrapper = shallow(
      <PrimeraApp saludo={saludo} subtitulo={subTitulo} />
    );

    const txtParrafo = wrapper.find("p").text();

    expect(txtParrafo).toBe(subTitulo);
    // console.log(txtParrafo);
  });
});
