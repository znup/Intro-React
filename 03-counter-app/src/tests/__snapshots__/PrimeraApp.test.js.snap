// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba de mi <PrimeraApp /> Debe mostrar <PrimeraApp /> correctamente 1`] = `
<Fragment>
  <h1>
    Un saludo mas
  </h1>
  <p>
    Es un subtitulo
  </p>
</Fragment>
`;
